Modules/_testcapimodule.c

```diff
@@ -2340,6 +2340,195 @@ get_timezone_utc_capi(PyObject* self, PyObject *args) {
     }
 }
 
+static PyObject *
+get_date_fromdate(PyObject *self, PyObject *args)
+{
+    PyObject *rv = NULL;
+    int year = 0, month = 0, day = 0;
+    int macro = 0;
+
+    if (!PyArg_ParseTuple(args, "iii|p", &year, &month, &day, &macro)) {
+        return NULL;
+    }
+
+    // Pass along to the API function
+    if (macro) {
+        rv = PyDate_FromDate(year, month, day);
+    }
+
+    else {
+        rv = PyDateTimeAPI->Date_FromDate(
+            year, month, day,
+            PyDateTimeAPI->DateType);
+    }
+
+    return rv;
+}
+
+static PyObject *
+get_datetime_fromdateandtime(PyObject *self, PyObject *args)
+{
+    PyObject *rv = NULL;
+    int year = 0, month = 0, day = 0;
+    int hour = 0, minute = 0, second = 0, microsecond = 0;
+    int macro = 0;
+
+    if (!PyArg_ParseTuple(
+        args,
+        "iiiiiii|p",
+        &year, &month, &day,
+        &hour, &minute, &second, &microsecond,
+        &macro))
+    {
+
+        return NULL;
+    }
+
+    // Pass along to the API function
+    if (macro) {
+        rv = PyDateTime_FromDateAndTime(
+            year, month, day,
+            hour, minute, second, microsecond);
+    }
+
+    else {
+        rv = PyDateTimeAPI->DateTime_FromDateAndTime(
+            year, month, day,
+            hour, minute, second, microsecond,
+            Py_None,
+            PyDateTimeAPI->DateTimeType);
+    }
+    return rv;
+}
+
+static PyObject *
+get_datetime_fromdateandtimeandfold(PyObject *self, PyObject *args)
+{
+    PyObject *rv = NULL;
+    int year = 0, month = 0, day = 0;
+    int hour = 0, minute = 0, second = 0, microsecond = 0, fold = 0;
+    int macro = 0;
+
+    if (!PyArg_ParseTuple(
```

Comment:
ditto, 
remove empty string and reformat your call to `PyArg_ParseTuple`
from PEP7 (https://www.python.org/dev/peps/pep-0007/)

Breaking long lines: if you can, break after commas in the outermost argument list. Always indent continuation lines appropriately, e.g.:

in this case, 
```c++
if (!PyArg_ParseTuple(args, "iiiiiiii|p",
                      &year, &month, &day,
                      &hour, &minute, &second, &microsecond,
                      &fold, &macro)) {
    return NULL;
}
```